app/services/permissions_checker.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class PermissionsChecker
+  def initialize(current_user, permission_name, params)
+    @current_user = current_user
+    @permission = Permission.find_by(name: permission_name)
```

Comment:
```suggestion
    @permission_name = permission_name
```

Reply:
This will save us an extra query (CHECK BELOW),

Reply:
Nice